tests/lib/utils.sh

```
@@ -457,3 +457,106 @@ setup_cleanup_trap() {
+
+    log_info "Waiting for compose services to be healthy (timeout: ${timeout}s)..."
+
+    while [ $elapsed -lt $timeout ]; do
+        local all_healthy=true
+        local services
+        services=$(docker compose $compose_files ps --format json 2>/dev/null | jq -sr '.[] | select(.Health != "healthy" and .Health != "" and .Health != null) | .Name' 2>/dev/null || true)
+
+        if [ -z "$services" ]; then
+            # Also check if any services are still starting (no healthcheck yet)
+            local starting
+            starting=$(docker compose $compose_files ps --format json 2>/dev/null | jq -sr '.[] | select(.State == "running" and (.Health == "" or .Health == null)) | .Name' 2>/dev/null || true)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)